rdkit/Chem/Draw/__init__.py

```diff
@@ -573,6 +571,238 @@ def _MolsToGridImage(mols, molsPerRow=3, subImgSize=(200, 200), legends=None,
+  else:
+    highlightBondLists = None
+
+  return mols, molsPerRow, legends, highlightAtomLists, highlightBondLists
+
+def MolsMatrixToGridImage(mols_matrix, subImgSize=(200, 200), legends_matrix = None, 
+                          highlightAtomLists_matrix = None, highlightBondLists_matrix = None, useSVG=False, returnPNG=False, **kwargs):
+  """Creates a mol grid image from a nested data structure (where each data substructure represents a row),
+  padding rows as needed so all rows are the length of the longest row
+          ARGUMENTS:
+
+        - mols_matrix: A two-deep nested data structure of RDKit molecules to draw,
+         for example list of lists of RDKit molecules
+
+        - subImgSize: The size of a cell in the drawing; passed through to MolsToGridImage (default (200, 200))
+
+        - legends_matrix: A two-deep nested data structure of strings to label molecules with,
+         for example list of lists of strings (default None)
+
+        - highlightAtomLists_matrix: A three-deep nested data structure of integers of atoms to highlight,
+         for example list of lists of lists of integers (default None)
+
+        - highlightBondLists_matrix: A three-deep nested data structure of integers of bonds to highlight,
+         for example list of lists of lists of integers (default None)
+
+        - useSVG: Whether to return an SVG (if true) or PNG (if false);
+         passed through to MolsToGridImage (default false)
+
+        - returnPNG: Whether to return a PNG file (if true) or a PIL object for a PNG image file (if false);
+         has no effect if useSVG is true; passed through to MolsToGridImage (default false)
+
+        - kwargs: Any other keyword arguments are passed to MolsToGridImage
+
+      NOTES:
+
+            To include a blank cell in the middle of a row, supply None for that entry in mols_matrix.
+            You do not need to do that for empty cells at the end of a row; 
+            this function will automatically pad rows so that all rows are the same length.
+            
+            This function is useful when each row has some meaning,
+            for example the generation in a mass spectrometry fragmentation tree--refer to 
+            example at https://en.wikipedia.org/wiki/Fragmentation_(mass_spectrometry).
+            If you want to display a set molecules where each row does not have any specific meaning,
+            use MolsToGridImage instead.
+
+            This function nests data structures one additional level beyond the analogous function MolsToGridImage
+            (in which the molecules and legends are non-nested lists, 
+            and the highlight parameters are two-deep nested lists) 
+
+      RETURNS:
+
+        A grid of molecular images in one of these formats:
+        
+        - useSVG=False and returnPNG=False (default): A PIL object for a PNG image file
+
+        - useSVG=False and returnPNG=True: A PNG file 
```

Comment:
it returns PNG data, not a file

Reply:
Thanks, these return types were difficult for me to determine